src/array.cr

```diff
@@ -1118,6 +1118,35 @@ class Array(T)
     nil
   end
 
+  def rotate!(n = 1)
+    return self if length == 0
+    n %= length if n.abs >= length
+    n += length if n < 0
```

Comment:
I would replace these two lines with
```crystal
n %= length
```
Maybe it would be slightly slower, maybe not, looks cleaner anyway.